src/bridged_logging.rs

```diff
@@ -0,0 +1,226 @@
+                GlibLogger::write_log_structured(
+                    domain,
+                    record.level(),
+                    record.file(),
+                    record.line(),
+                    None,
+                    &format!("{}", record.args()),
+                );
+            },
+        };
+    }
+
+    fn flush(&self) {}
+}
+
+#[macro_export]
+#[cfg(any(feature = "dox", feature = "bridged_logging_domain_macros"))]
+/// A macro which behaves exactly as `log::error!` except that it sets the
+/// current log target to the contents of a `G_LOG_DOMAIN` constant (and fails
+/// to build if not defined).
+macro_rules! error {
+    (target: $target:expr, $($arg:tt)+) => (
+        log::log!(target: $target, log::Level::Error, $($arg)+);
+    );
+    ($($arg:tt)+) => (
+        log::log!(target: G_LOG_DOMAIN, log::Level::Error, $($arg)+);
+    )
+}
+
+#[macro_export]
+#[cfg(any(feature = "dox", feature = "bridged_logging_domain_macros"))]
+/// A macro which behaves exactly as `log::warn!` except that it sets the
+/// current log target to the contents of a `G_LOG_DOMAIN` constant (and fails
+/// to build if not defined).
+macro_rules! warn {
+    (target: $target:expr, $($arg:tt)+) => (
+        log::log!(target: $target, log::Level::Warn, $($arg)+);
+    );
+    ($($arg:tt)+) => (
+        log::log!(target: G_LOG_DOMAIN, log::Level::Warn, $($arg)+);
+    )
+}
+
+#[macro_export]
+#[cfg(any(feature = "dox", feature = "bridged_logging_domain_macros"))]
+/// A macro which behaves exactly as `log::info!` except that it sets the
+/// current log target to the contents of a `G_LOG_DOMAIN` constant (and fails
+/// to build if not defined).
+macro_rules! info {
+    (target: $target:expr, $($arg:tt)+) => (
+        log::log!(target: $target, log::Level::Info, $($arg)+);
+    );
+    ($($arg:tt)+) => (
+        log::log!(target: G_LOG_DOMAIN, log::Level::Info, $($arg)+);
+    )
+}
+
+#[macro_export]
+#[cfg(any(feature = "dox", feature = "bridged_logging_domain_macros"))]
+/// A macro which behaves exactly as `log::debug!` except that it sets the
+/// current log target to the contents of a `G_LOG_DOMAIN` constant (and fails
+/// to build if not defined).
+macro_rules! debug {
+    (target: $target:expr, $($arg:tt)+) => (
+        log::log!(target: $target, log::Level::Debug, $($arg)+);
+    );
+    ($($arg:tt)+) => (
+        log::log!(target: G_LOG_DOMAIN, log::Level::Debug, $($arg)+);
+    )
+}
+
+#[macro_export]
+#[cfg(any(feature = "dox", feature = "bridged_logging_domain_macros"))]
+/// A macro which behaves exactly as `log::trace!` except that it sets the
+/// current log target to the contents of a `G_LOG_DOMAIN` constant (and fails
+/// to build if not defined).
+macro_rules! trace {
+    (target: $target:expr, $($arg:tt)+) => (
+        log::log!(target: $target, log::Level::Trace, $($arg)+);
+    );
+    ($($arg:tt)+) => (
+        log::log!(target: G_LOG_DOMAIN, log::Level::Trace, $($arg)+);
```

Comment:
I didn't know that the macro hygiene rules allowed this, interesting.

But can you put these macros into a separate commit in the same PR?

Reply:
Sure, I'll do one forced commit without and another with.